src/m365/spo/commands/group/group-user-add.ts

```diff
@@ -76,11 +76,25 @@ class SpoGroupUserAddCommand extends SpoCommand {
       }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
   }
 
-  private getGroupId(args: CommandArgs): Promise<number> {
```

Comment:
To avoid duplicating the code for issuing the request and parsing the response, we could put both methods in one like:

```ts
  private getGroupId(args: CommandArgs): Promise<number> {
    const getGroupMethod: string = args.options.groupName ? 
      `GetByName('${encodeURIComponent(args.options.groupName as string)}')` :
      `GetById('${args.options.groupId}')`;

    const requestOptions: any = {
      url: `${args.options.webUrl}/_api/web/sitegroups/${getGroupMethod}`,
      headers: {
        'accept': 'application/json;odata=nometadata'
      },
      responseType: 'json'
    };

    return request
      .get<{ Id: number }>(requestOptions)
      .then(response => {
        const groupId: number | undefined = response.Id;

        if (!groupId) {
          return Promise.reject(`The specified group does not exist in the SharePoint site`);
        }

        return groupId;
      });
  }
```